src/api/Migrations/20230417152352_AddWaterIngressModel.cs

```diff
@@ -0,0 +1,161 @@
+using Npgsql.EntityFrameworkCore.PostgreSQL.Metadata;
+
+#nullable disable
+
+namespace BDMS.Migrations;
+
+public partial class AddWaterIngressModel : Migration
+{
+    protected override void Up(MigrationBuilder migrationBuilder)
+    {
+        migrationBuilder.CreateTable(
+            name: "observation",
+            schema: "bdms",
+            columns: table => new
+            {
+                id = table.Column<int>(type: "integer", nullable: false)
+                    .Annotation("Npgsql:ValueGenerationStrategy", NpgsqlValueGenerationStrategy.IdentityByDefaultColumn),
+                observation_type = table.Column<int>(type: "integer", nullable: false),
+                start_time = table.Column<DateTime>(type: "timestamp with time zone", nullable: true),
+                end_time = table.Column<DateTime>(type: "timestamp with time zone", nullable: true),
+                duration = table.Column<double>(type: "double precision", nullable: true),
+                from_depth_m = table.Column<double>(type: "double precision", nullable: true),
+                to_depth_m = table.Column<double>(type: "double precision", nullable: true),
+                from_depth_masl = table.Column<double>(type: "double precision", nullable: true),
+                to_depth_masl = table.Column<double>(type: "double precision", nullable: true),
+                completion_finished = table.Column<bool>(type: "boolean", nullable: true),
+                casing = table.Column<int>(type: "integer", nullable: true),
+                comment = table.Column<string>(type: "text", nullable: true),
+                reliability = table.Column<int>(type: "integer", nullable: false),
+                borehole_id = table.Column<int>(type: "integer", nullable: false),
+                creator = table.Column<int>(type: "integer", nullable: true),
+                creation = table.Column<DateTime>(type: "timestamp with time zone", nullable: true),
+                updater = table.Column<int>(type: "integer", nullable: true),
+                update = table.Column<DateTime>(type: "timestamp with time zone", nullable: true),
+            },
+            constraints: table =>
+            {
+                table.PrimaryKey("PK_observation", x => x.id);
+                table.ForeignKey(
+                    name: "FK_observation_borehole_borehole_id",
+                    column: x => x.borehole_id,
+                    principalSchema: "bdms",
+                    principalTable: "borehole",
+                    principalColumn: "id_bho",
+                    onDelete: ReferentialAction.Cascade);
+                table.ForeignKey(
+                    name: "FK_observation_codelist_reliability",
+                    column: x => x.reliability,
+                    principalSchema: "bdms",
+                    principalTable: "codelist",
+                    principalColumn: "id_cli",
+                    onDelete: ReferentialAction.Cascade);
```

Comment:
REQ: Ich denke hier dürfen wir ein Delete nicht kaskadieren, ansonsten werden unter Umständen Codelist-Einträge gelöscht. Oder verstehe ich etwas falsch?

Reply:
Ich denke es ist andersherum, dass wenn der Codelist Eintrag gelöscht wird, werden die entsprechenden Observations gelöscht, was weniger ein Problem sein sollte als andersherum. Aber wir können es trotzdem entfernen.
 Ich verstehe nicht ganz nach welcher Logik beim automatisierten Erstellen der Migrationen "OnDelete Cascade" gesetzt wird. Beim User haben wir es hier Beispielsweise nicht, in anderen Tables wurde das zunächst gesetzt und dann haben wir es in einer weiteren Migration entfernt soweit ich verstehen.

Reply:
Stimmt, sorry das habe ich verwechselt. Die Aussage stimmt aber weiterhin: Beim Löschen von Codelist-Werten möchten wir nicht, dass automatisch Observation-Einträge gelöscht werden. In diesem Fall möchten wir einen Fehler in der Datenbank erhalten, der uns sagt, dass der Codelist-Wert noch verwendet wird.